bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandlerCallback.java

```diff
@@ -115,6 +115,18 @@
      */
     ChannelBuilder createChannelBuilder(ChannelUID channelUID, ChannelTypeUID channelTypeUID);
 
+    /**
+     * Creates a {@link ChannelBuilder} which is preconfigured with values from the given channel and allows to modify
+     * it. The methods {@link BaseThingHandler#editThing(Thing)} and {@link BaseThingHandler#updateThing(Thing)} must be
+     * called to persist the changes.
+     *
+     * @param thing thing (must not be null)
+     * @param channelUID the UID of the channel to be edited
+     * @return a preconfigured ChannelBuilder
+     * @throw {@link IllegalArgumentException} if no channel with the given UID exists for the given thing
+     */
+    ChannelBuilder editChannelBuilder(Thing thing, ChannelUID channelUID);
```

Comment:
The method name "feels" a little bit strange. I know its consistent with `createChannelBuilder`. But `edit ChannelBuilder` neither accepts a ChannelBuilder parameter nor does it edit an existing ChannelBuilder.
How about a simple `editChannel`? Since Channel is immutable now the return type is `ChannelBuilder` which in turn is self-explanatory. wdyt?

Reply:
I agree, struggeled with this too. `editChannel` sounds better.